packages/opencensus-core/src/trace/sampler/sampler.ts

```diff
@@ -97,3 +99,40 @@ export class SamplerBuilder {
     return new ProbabilitySampler(probability);
   }
 }
+
+/** Builder Class of TraceParams */
+export class TraceParamsBuilder {
+  private static readonly MAX_NUMBER_OF_ANNOTATION_EVENTS_PER_SPAN = 32;
+  private static readonly MAX_NUMBER_OF_MESSAGE_EVENTS_PER_SPAN = 128;
+  private static readonly MAX_NUMBER_OF_ATTRIBUTES_PER_SPAN = 32;
+  private static readonly MAX_NUMBER_OF_LINKS_PER_SPAN = 32;
+
+  static getNumberOfAnnotationEventsPerSpan(traceParameters: TraceParams):
```

Comment:
What would you think about making these functions be top level `function` exports rather than static members of a class? You could still namespace them when they are used if needed based on how they are imported.

Reply:
all four functions are a part of TraceParams, right? so exporting them individually out of sampler.ts doesn't seem right. Instead, we are exporting them under TraceParams class. Just like in the specs document https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/TraceConfig.md#traceparams.

Reply:
Also since we are providing default values, in case user doesn't provide them, so are those functions really worth exporting?

Reply:
Yeah, I would be OK exporting these under a containing class in this case.

And having separate sets of constants for the defaults and maximums would be OK too I think. I'll modify my comment above to just ask that we make them top level constants for the defaults instead of properties.